### ISSUE TITLE
R code inside header-includes is no longer evaluated

### ISSUE THREAD

**Issue by mcol** [rstudio/rmarkdown#1709]
Since 1.18, R code inside `header-includes` is not evaluated, but outside of it is.

Reproducible example (succeeds with 1.17, fails with 1.18):
```
---
title: "Issue"
author: "mcol"
header-includes:
- \externaldocument{`r sprintf("file_%s", format(Sys.Date(), "%Y-%m-%d"))`}
date: "`r format(Sys.Date(), '%-d %B %Y')`"
output: pdf_document
---

Test
```

Error message:
```
! Undefined control sequence.
l.66 \externaldocument
                      {`r sprintf("file_%s", format(Sys.Date(), "%Y-%m-%d"))`}

Error: Failed to compile test.tex.
```

Session info:
```
R version 3.6.0 (2019-04-26)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.4 LTS

Locale:
  LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C
  LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8
  LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8
  LC_PAPER=en_GB.UTF-8       LC_NAME=C
  LC_ADDRESS=C               LC_TELEPHONE=C
  LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C

Package version:
  base64enc_0.1.3 digest_0.6.23   evaluate_0.14   glue_1.3.1
  graphics_3.6.0  grDevices_3.6.0 highr_0.8       htmltools_0.4.0
  jsonlite_1.6    knitr_1.26      magrittr_1.5    markdown_1.1
  methods_3.6.0   mime_0.7        Rcpp_1.0.3      rlang_0.4.2
  rmarkdown_1.18  stats_3.6.0     stringi_1.4.3   stringr_1.4.0
  tinytex_0.17    tools_3.6.0     utils_3.6.0     xfun_0.11
  yaml_2.2.0

Pandoc version: 1.17.2
```
---

By filing an issue to this repo, I promise that

- [X] I have fully read the issue guide at https://yihui.org/issue/.
- [X] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('rmarkdown')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('rstudio/rmarkdown')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [X] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.

**Comment by cderv** [comment]
This happened in 5499ec63dbf4b15eb8769e4a0c59843ac53abfd2 to fix #1359

Before, `header-include` was passed to pandoc through yaml header and when converting `.md` to `.tex` the r code was executed. 

The change mentioned above write the header-include to a file to pass in pandoc using command line.  It means the the r code in `header-include` is wrote as-is in and not currently knit when convert to md. 

If we look at the issue https://github.com/rstudio/rmarkdown/issues/1359#issuecomment-464707364, it was already the case when setting `extra_dependency`

This is a bug I think - or a least a feature the could be added.

**Comment by yihui** [comment]
Should be fixed now. Thanks a lot for the report!
